### ISSUE TITLE
Drag and Drop Functionality, How can I disable this feature?

### ISSUE THREAD

**Issue by CarmanML** [jupyter/help#82]
Hello, I am setting up an environment for some users in which I need to narrow the capabilites.  Specifically i need to make sure no external content can be uploaded into Jupyter Notebook.  I was able to locate the code and eliminate the "Upload" button.

How or where can I modify code to disable the drag and drop?    
Many thanks in advance!
Mike

**Comment by gnestor** [comment]
@Carreau @takluyver What would be the most-straightforward way to disable the drag'n'drop extension? In `custom.js`? I was looking at the [extension source](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/master/src/jupyter_contrib_nbextensions/nbextensions/dragdrop/main.js) and you could just reverse the changes it makes, but I have a feeling there is a better way...

**Comment by takluyver** [comment]
Disable the extension and reload the page? We don't have a specific way for extensions to be unloaded without reloading the page.

**Comment by gnestor** [comment]
Right, so @CarmanML, when creating your Jupyter environment just make sure that the drag'n'drop extension is disabled (I think it's enabled by default): `jupyter nbextension disable dragdrop`. I've tested and verified that when disabled, dragging an image into a notebook will navigate to its local URL and not upload it to the server.

**Comment by CarmanML** [comment]
Awesome!  Thank you so much for the support.

Cheers,
Mike

**Comment by takluyver** [comment]
Hang on, looking back at this, I think you were talking at cross purposes. @gnestor meant the [drag and drop extension](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/b29c698394239a6931fa4911440550df214812cb/src/jupyter_contrib_nbextensions/nbextensions/dragdrop) in ipython_contrib. @CarmanML most likely meant dragging and dropping files into the dashboard to upload them. That's not part of an extension, and I don't think there's an easy way to disable it.

@CarmanML can you tell us a bit more about what you're trying to achieve? Jupyter is designed to let the user do anything in the environment it runs in, so it's generally quite hard to seriously restrict what users can do. A better approach is usually to run it in an environment that the user is allowed to mess up, e.g. a Docker container.
